### ISSUE TITLE
Move (Re)Provider logic to Content Routers

### ISSUE THREAD

**Issue by guillaumemichel** [ipfs/go-ipfs-provider#49]
The Reprovide logic is usually specific to the Content Router(s) in use. Hence, a `Provider` interface should be exposed through [go-libp2p-routing-helpers](https://github.com/libp2p/go-libp2p-routing-helpers), and the reprovide logic should move away from this repo to the specific Content Routers (e.g DHT, IPNI).

See:
* https://github.com/libp2p/go-libp2p-kad-dht/issues/824
* https://github.com/libp2p/go-libp2p/issues/2175

**Comment by guseggert** [comment]
Related to https://github.com/ipfs/kubo/issues/9682 and https://github.com/ipfs/kubo/issues/9482

**Comment by hacdias** [comment]
This repository is no longer maintained and has been copied over to [Boxo](https://github.com/ipfs/boxo).  In an effort to avoid noise and crippling in the Boxo repo from the weight of issues of the past, we are closing most issues and PRs in this repo.  Please feel free to open a new issue in Boxo (and reference this issue) if resolving this issue is still critical for unblocking or improving your usecase.

You can learn more in the [FAQs for the Boxo repo copying/consolidation effort](https://github.com/ipfs/boxo/wiki/Copied-or-Migrated-Repos-FAQ).
